syntax = "proto3";

package bidirectional;

class UserDetails {
  int age;
  int weight;
  int lenght;
}

service Bidirectional {
  // A Bidirectional streaming RPC.
  //
  // Accepts a stream of Message sent while a route is being traversed,
   rpc GetServerResponse(stream Message) returns (stream Message) {}
}

message Message {
    optional string action = 1;
    optional string client_username = 2;
    optional string client_password = 3;
    optional string recipient_username = 4;
    optional string message = 5;
    optional list available_messages = 6;
    optional list list_accounts = 7;
}
